trade-finance-manager-api/src/v1/controllers/amendment.controller.js

```diff
@@ -179,13 +181,24 @@ const getAllAmendments = async (req, res) => {
   return res.status(422).send({ data: 'Unable to fetch amendments' });
 };
 
-const createFacilityAmendment = async (req, res) => {
-  const { facilityId } = req.body;
-  const { amendmentId } = await api.createFacilityAmendment(facilityId, generateTfmAuditDetails(req.user._id));
-  if (amendmentId) {
+const createFacilityAmendment = async (req, res, next) => {
+  try {
+    const { facilityId } = req.body;
+
+    const { amendmentId } = await api.createFacilityAmendment(facilityId, generateTfmAuditDetails(req.user._id));
+
+    if (!amendmentId) {
+      throw new AmendmentNotCreatedError(facilityId);
+    }
+
     return res.status(200).send({ amendmentId });
+  } catch (error) {
+    if (isAxiosError(error)) {
+      return res.status(error?.response?.status || 500).json({ data: 'Failed to create facility amendment' });
+    }
+
+    return next(error);
```

Comment:
This handles all our apiErrors automatically for us now with global middleware

Reply:
Does `throw error` not work?

Reply:
Seen your comment elsewhere now this makes sense plz ignore